src/nuclide_header.F90

```diff
@@ -0,0 +1,761 @@
+  end type NuclideBase
+
+  abstract interface
+
+    subroutine print_nuclide_(this, unit)
+      import NuclideBase
+      class(NuclideBase),intent(in) :: this
+      integer, optional,  intent(in) :: unit
+    end subroutine print_nuclide_
+
+  end interface
+
+  type, extends(NuclideBase) :: NuclideCE
+    ! Energy grid information
+    integer :: n_grid                     ! # of nuclide grid points
+    integer, allocatable :: grid_index(:) ! log grid mapping indices
+    real(8), allocatable :: energy(:)     ! energy values corresponding to xs
+
+    ! Microscopic cross sections
+    real(8), allocatable :: total(:)      ! total cross section
+    real(8), allocatable :: elastic(:)    ! elastic scattering
+    real(8), allocatable :: fission(:)    ! fission
+    real(8), allocatable :: nu_fission(:) ! neutron production
+    real(8), allocatable :: absorption(:) ! absorption (MT > 100)
+    real(8), allocatable :: heating(:)    ! heating
+
+    ! Resonance scattering info
+    logical              :: resonant = .false. ! resonant scatterer?
+    character(10)        :: name_0K = '' ! name of 0K nuclide, e.g. 92235.00c
+    character(16)        :: scheme ! target velocity sampling scheme
+    integer              :: n_grid_0K ! number of 0K energy grid points
+    real(8), allocatable :: energy_0K(:)  ! energy grid for 0K xs
+    real(8), allocatable :: elastic_0K(:) ! Microscopic elastic cross section
+    real(8), allocatable :: xs_cdf(:) ! CDF of v_rel times cross section
+    real(8)              :: E_min ! lower cutoff energy for res scattering
+    real(8)              :: E_max ! upper cutoff energy for res scattering
+
+    ! Fission information
+    logical :: has_partial_fission ! nuclide has partial fission reactions?
+    integer :: n_fission           ! # of fission reactions
+    integer, allocatable :: index_fission(:) ! indices in reactions
+
+    ! Total fission neutron emission
+    integer :: nu_t_type
+    real(8), allocatable :: nu_t_data(:)
+
+    ! Prompt fission neutron emission
+    integer :: nu_p_type
+    real(8), allocatable :: nu_p_data(:)
+
+    ! Delayed fission neutron emission
+    integer :: nu_d_type
+    integer :: n_precursor ! # of delayed neutron precursors
+    real(8), allocatable :: nu_d_data(:)
+    real(8), allocatable :: nu_d_precursor_data(:)
+    type(AngleEnergyContainer), allocatable :: nu_d_edist(:)
+
+    ! Unresolved resonance data
+    logical                :: urr_present
+    integer                :: urr_inelastic
+    type(UrrData), pointer :: urr_data => null()
+
+    ! Reactions
+    integer :: n_reaction ! # of reactions
+    type(Reaction), allocatable :: reactions(:)
+    type(DictIntInt) :: reaction_index ! map MT values to index in reactions
+                                       ! array; used at tally-time
+
+    ! Type-Bound procedures
+    contains
+      procedure, pass :: clear => nuclidece_clear
+      procedure, pass :: print => nuclidece_print
```

Comment:
Move `contains` and procedure definitions two spaces back. Remove superfluous `pass` specifications.